test/python/quantum_info/test_sparse_observable.py

```diff
@@ -2335,6 +2335,21 @@ def test_compose_qargs_failure_duplicates(self):
         with self.assertRaisesRegex(ValueError, "duplicate indices in qargs"):
             SparseObservable.identity(5).compose("XYZX", qargs=[0, 1, 1, 0])
 
+    @ddt.data(
+        ("Y", "Y", True),
+        ("Y", "Z", False),
+        ("XX", "ZZ", True),
+        ("0XX", "0ZZ", True),
+        ("1XX", "0ZZ", True),
+        ("+XX", "0ZZ", False),
+    )
+    @ddt.unpack
+    def test_commute(self, p1, p2, expected):
+        """Test commutation"""
+        op1 = SparseObservable(p1)
+        op2 = SparseObservable(p2)
+        self.assertEqual(op1.commutes(op2), expected)
```

Comment:
this test is of a SparseObservable with a single Pauli term, what happens if there are several terms and coefficients?

Reply:
I can add some